npc_gzip/knn_classifier.py

```
@@ -0,0 +1,322 @@
+        ).reshape(-1)
+
+    def _calculate_distance(
+        self,
+        compressed_x: np.ndarray,
+        compressed_combined: np.ndarray,
+        compressed_training: np.ndarray = None,
+    ) -> np.ndarray:
+        """
+        Helper function that converts the string representation
+        of `self.distance_metric` to the actual
+        `npc_gzip.distance.Distance.[distance_metric]`. Then
+        that distance metric is calculated using
+        `self.compressed_training_inputs`, `compressed_x` and
+        `compressed_combined`.
+
+        Arguments:
+            compressed_x (np.ndarray): Numpy array representing the
+                                       compressed lengths of the input
+                                       data to be scored.
+            compressed_combined (np.ndarray): Numpy array representing the
+                                              compressed lengths of the input
+                                              data combined with
+                                              each training sample to be scored.
+        Returns:
+            np.ndarray: Numpy array containing the distance metric.
+        """
+
+        if compressed_training is None:
+            distance = Distance(
+                np.resize(self.compressed_training_inputs, compressed_x.shape),
+                compressed_x,
+                compressed_combined,
+            )
+        else:
+            distance = Distance(
+                np.resize(compressed_training, compressed_x.shape),
+                compressed_x,
+                compressed_combined,
+            )
+
+        if self.distance_metric == "ncd":
+            return distance.ncd
+        elif self.distance_metric == "clm":
+            return distance.clm
+        elif self.distance_metric == "cdm":
+            return distance.cdm
+        elif self.distance_metric == "mse":
+            return distance.mse
+        else:
+            return "Invalid Distance Metric"
```

This should raise an `UnsupportedDistanceMetricException` exception rather than *returning* the error message.